torch/onnx/symbolic_opset13.py

```diff
@@ -539,6 +539,58 @@ def unsafe_chunk(g: jit_utils.GraphContext, self, chunks, dim, _outputs=None):
     return g.op("Split", self, splits, axis_i=dim, outputs=_outputs)
 
 
+@_onnx_symbolic("aten::tile")
+@_beartype.beartype
+def tile(g: jit_utils.GraphContext, self, dims):
+    self_shape = g.op("Shape", self)
+    self_rank = g.op("Size", self_shape)
+    dims_rank = g.op("Size", dims)
+    diff = g.op("Sub", self_rank, dims_rank)
+    const_zero = g.op("Constant", value_t=torch.tensor([0]))
+
+    # 1. If dims is shorter than self.shape pad dims with 1
+    cond_greater = g.op("Greater", diff, const_zero)
+    (
+        if_op_greater,
+        (if_context_greater, else_context_greater),
+        _,
+    ) = jit_utils.add_op_with_blocks(g, "If", cond_greater, n_blocks=2, outputs=1)
+    const_one = if_context_greater.op("Constant", value_t=torch.LongTensor([1]))
+    diff_1d_greater = if_context_greater.op("Reshape", diff, const_one)
+    exapnd_ones_greater = if_context_greater.op("Expand", const_one, diff_1d_greater)
+    dims_ = if_context_greater.op("Concat", exapnd_ones_greater, dims, axis_i=0)
+    utils._add_output_to_block(if_context_greater.block, dims_)
+    identity_dim = else_context_greater.op("Identity", dims)
+    utils._add_output_to_block(else_context_greater.block, identity_dim)
+    dims_final = if_op_greater.node().output()
+
+    # 2. If dims is longer than self.shape pad self.shape with 1
+    cond_lesser = g.op("Less", diff, const_zero)
+    (
+        if_op_lesser,
+        (if_context_lesser, else_context_lesser),
+        _,
+    ) = jit_utils.add_op_with_blocks(g, "If", cond_lesser, n_blocks=2, outputs=1)
+    const_one = if_context_lesser.op("Constant", value_t=torch.LongTensor([1]))
+    diff_1d_lesser = if_context_lesser.op(
+        "Reshape",
+        if_context_lesser.op("Abs", diff),
+        const_one,
+    )
+    exapnd_ones_lesser = if_context_lesser.op("Expand", const_one, diff_1d_lesser)
```

Comment:
Here and elsewhere: consider more descriptive variable names. Otherwise lesser -> less